nibabel/loadsave.py

```diff
@@ -105,25 +83,38 @@ def save(img, filename):
     -------
     None
     '''
+
+    # Save the type as expected
     try:
         img.to_filename(filename)
     except ImageFileError:
         pass
     else:
         return
-    froot, ext, trailing = splitext_addext(filename, ('.gz', '.bz2'))
+
+    # Be nice to users by making common implicit conversions
+    froot, ext, trailing = splitext_addext(filename, img._compressed_exts)
```

Comment:
Yeah, I think this was a mistake that assumed there wouldn't be a conversion. We'll just have to update this list with any new compression extensions that get used, which seems unlikely enough.